statics/js/components/rule-detail.js

```diff
@@ -0,0 +1,494 @@
+   * @param the action to represent
+   * @return path of the icon to use.
+  */
+  BridgeLayout.prototype.imgsrc = function(act) {
+    var file;
+    switch (act) {
+      case ActionOutput:
+        file = 'output.png';
+        break;
+      case ActionResubmit:
+        file = 'resubmit.png';
+        break;
+      case ActionFlood:
+        file = 'flood.png';
+        break;
+      case ActionNormal:
+        file = 'flood.png';
+        break;
+      case ActionDrop:
+        file = 'drop.png';
+        break;
+      default:
+        return null;
+    }
+    return 'statics/img/openflow/' + file;
+  };
+
+  /** Readable name of an openflow port
+   * @param pr the index of the port or its stringified value.
+  */
+  BridgeLayout.prototype.portname = function(pr) {
+    var p = (typeof (pr) === 'string') ? parseInt(pr) : pr;
+    if (p === ANY_PORT) return 'ANY';
+    if (p === LOCAL_PORT) return 'LOCAL';
+    if (p === SAME_PORT || p === undefined) return '';
+    var nodes = this.interfaces[p];
+    if (nodes === undefined) return '???';
+    var port = nodes[1];
+    var portname = (port === undefined) ? '???' : port.metadata.Name;
+    return portname;
+  };
+
+  /** Check if rule is highlighted. */
+  BridgeLayout.prototype.isHighlighted = function(rule) {
+    var current = this.store.state.currentRule;
+    var status = current && rule.UUID === current.metadata.UUID;
+    return status;
+  };
+
+  return BridgeLayout;
+}());
+
+/** Graphical component that represents a set of rules associated to a given
+ *  Openflow table and port and displayed as a single HTML table.
+ */
+Vue.component('rule-table-detail', {
+  template: '\
+    <div class="dynamic-table">\
+      <table class="table table-bordered table-condensed">\
+        <thead>\
+            <tr>\
+                <th class="filters-column">filters</th>\
+                <th class="summary-column">summary</th>\
+                <th class="actions-column">actions</th>\
+            </tr>\
+        </thead>\
+        <tbody>\
+            <tr v-for="rule in rules"\
+                :id="\'R-\' + rule.UUID"\
+                v-bind:class="{soft: layout.isHighlighted(rule)}">\
+                <td>{{rule.filters}}</td>\
+                <td>\
+                    <table>\
+                        <tr v-for="act in rule.outAction">\
+                            <td><img :src="layout.imgsrc(act.action)"\
+                                    width="16"\
+                                    height="16" /></td>\
+                            <td v-on:mouseover="layout.mark(act.port)"\
+                                v-on:mouseleave="layout.unmark(act.port)"\
+                                v-on:click="layout.select(act.port)">\
+                                <span class="port-link">{{layout.portname(act.port)}}</span>\
+                            </td>\
+                            <td> <a class="table-link" v-on:click="layout.switchTab(act.table)">{{act.table}}</a></td>\
```

Comment:
Indeed. Lost with the change from nav-tabs to nav-pills as the class was used to find the relevant link to select.